File: SmartInventoryFrontend/src/types/inventory.ts
import { Entity } from './common'
import { Item } from './item'
import { Location } from './location'

export interface Inventory extends Entity {
  ID: number
  Item_ID: number
  Location_ID: number
  OnHand_Quantity: number
  Available_Quantity: number
  Item_Name?: string
  Location_Name?: string
}

export interface AdjustInventoryRequest {
  Item_ID: number
  Location_ID: number
  QuantityAdjustment: number
  Remark?: string
}

export interface StockTransferRequest {
  From_Location_ID: number
  To_Location_ID: number
  Item_ID: number
  Transfer_Quantity: number
  Remark?: string
}

export interface InventoryFilters {
  skip: number
  take: number
  locationId?: number
  itemId?: number
  lowStockOnly?: boolean
}

export interface InventoryState {
  inventory: Inventory[]
  currentInventory: Inventory | null
  loading: boolean
  error: string | null
  total: number
  skip: number
  take: number
}

File: SmartInventoryAPI/Models/Entities/Inventory.cs
namespace SmartInventoryAPI.Models.Entities;

public class Inventory
{
    public long ID { get; set; }
    public long Item_ID { get; set; }
    public long Location_ID { get; set; }
    public decimal On_Hand_Quantity { get; set; }
    public decimal Available_Quantity { get; set; }
    public bool Is_Deleted { get; set; } = false;

    // Foreign keys
    public virtual Item? Item { get; set; }
    public virtual Location? Location { get; set; }
}
